common/src/main/java/tachyon/worker/WorkerClient.java

```diff
@@ -325,8 +312,12 @@ public synchronized String lockBlock(long blockId) throws IOException {
 
     try {
       return mClient.lockBlock(blockId, mSessionId);
-    } catch (FileDoesNotExistException e) {
-      return null;
+    } catch (TachyonTException e) {
+      if (e.getType().equals(TachyonExceptionType.FILE_DOES_NOT_EXIST.name())) {
```

Comment:
It would be nice to use a design that uses a single method for the:

```
mustConnect();
try {
  ...
} catch (TachyonTException e) {
 throw new TachyonException(e)
} catch (TException e) {
  mConnected = false;
}
```

boilerplate, but lets leave that for another PR. Perhaps add a TODO(jiri) for this here.